### ISSUE TITLE
Support Fedora (CentOS/RedHat)

### ISSUE THREAD

**Issue by abitrolly** [open-eid/linux-installer#13]
I'd like to [update my certificate](https://medium.com/e-residency-blog/estonia-is-enhancing-the-security-of-its-digital-identities-361b9a3c9c52) from Fedora. There are no packages, no build files for, it and no Makefile to build the tool myself.

**Comment by abitrolly** [comment]
From description at https://github.com/open-eid/linux-installer/blob/master/debian/control it looks like the primary tool that I need is [qesteidutil](https://github.com/open-eid/qesteidutil).

**Comment by metsma** [comment]
Feel free to help upstream package maintainers
https://admin.fedoraproject.org/pkgdb/package/rpms/qesteidutil/

**Comment by abitrolly** [comment]
It is not packaged for latest Fedora releases - https://apps.fedoraproject.org/packages/qesteidutil - looks like this [open source support plan](https://medium.com/@nayafia/how-i-stumbled-upon-the-internet-s-biggest-blind-spot-b9aa23618c58) is not very effective. )

**Comment by vodolaz095** [comment]
i'll be grateful, if release for Fedora 26 would be present!

**Comment by gitcoinbot** [comment]
__This issue now has a funding of 0.045 ETH (21.76 USDT) attached to it.__

 * If you would like to work on this issue you can claim it [here](https://gitcoin.co/funding/details?url=https://github.com/open-eid/linux-installer/issues/13).
 * If you've completed this issue and want to claim the bounty you can do so [here](https://gitcoin.co/funding/details?url=https://github.com/open-eid/linux-installer/issues/13)

**Comment by owocki** [comment]
hi from gitcoin.co -- doing my best to source someone to turn this around.  will keep yall posted

**Comment by owocki** [comment]
still working on it..

**Comment by step21** [comment]
I am quite familiar with debian packages, but still seems like a lot of work for that bounty. Have you tried to use alien https://sourceforge.net/projects/alien-pkg-convert/ It should also be available in fedora...

**Comment by gitcoinbot** [comment]
__The funding of 0.045 ETH (33.2 USD) attached has been claimed by @muhammaddava1321.__ 

 @muhammaddava1321, please leave a comment to let the funder (@abitrolly) and the other parties involved your implementation plan.  If you don't leave a comment, the funder may expire your claim at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/funding/details?url=https://github.com/open-eid/linux-installer/issues/13)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $11701.18 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by abitrolly** [comment]
@step21 you can find former work here - https://admin.fedoraproject.org/pkgdb/package/rpms/qesteidutil/

**Comment by abitrolly** [comment]
Ad for bounty, this is all I can get for this issue. If you can convince guys from this project to set a maintenance fund for different package maintainers - that will help to increase the funding.

**Comment by step21** [comment]
@abitrolly thanks. ok that would make it easier. Did you try just installing the older package? Somtimes that works, esp version jump from 25 to 26 is not that big. Just trying this before installing fedora packaging tools etc...

**Comment by owocki** [comment]
@muhammaddava1321 mind popping in and letting us nkow what we can expect from you/

**Comment by owocki** [comment]
@muhammaddava1321 if we dont hear from you soon we will need to reject your claim so that someone who can make this a prioroity takes the issue

**Comment by abitrolly** [comment]
@step21 what is the proper way to install older package?
```
~ sudo dnf install qesteidutil
Last metadata expiration check: 0:09:18 ago on Fri 29 Dec 2017 12:28:01 AM +03.
No match for argument: qesteidutil
Error: Unable to find a match
```

**Comment by step21** [comment]
Mmmh, looks like it is not available anymore on fedora pkgdb. What you can anyway try - download the newest one here from rpmfind, which is 3.12 and for fedora 25 should be new enough http://rpmfind.net/linux/rpm2html/search.php?query=qesteidutil(x86-64)
Then install with `sudo yum localinstall <path to downloaded package` Obviously replace the <> with the path to the package you downloaded and the package name. If that doesn't work, it should not be too hard to build from the source package, because that is still available at https://admin.fedoraproject.org/pkgdb/package/rpms/qesteidutil/ http://pkgs.fedoraproject.org/cgit/rpms/qesteidutil.git/

**Comment by abitrolly** [comment]
There is no point to run the old version that doesn't fix problem with hacked ids. How to build it from source?

**Comment by step21** [comment]
Mmh ok sorry it looked fairly recent to me at first.
Clone the soure package from http://pkgs.fedoraproject.org/cgit/rpms/qesteidutil.git/ then build an rpm from that. Maybe I can do it tonight if I don't do anything else.

**Comment by owocki** [comment]
i am rejecting @muhammaddava1321's claim on this issue due to radio silence from him/her

**Comment by owocki** [comment]
@abitrolly i suggest rejecting @muhammaddava1321's claim on this issue due to radio silence from him/her.  lets open up the issue for someone else to claim. 

you can do so at https://gitcoin.co/funding/details?url=https://github.com/open-eid/linux-installer/issues/13

**Comment by step21** [comment]
I started a build project on open build service, but so far it only builds successfully on fedora 25 because of some patch problem.
However I also saw that some other guy published a temporary repo. You have to put it into fedora software sources I guess, I couldn't test it. Maybe that works for you.
https://github.com/open-eid/qesteidutil/issues/78#issuecomment-347653717

**Comment by step21** [comment]
You can now find my build here: https://build.opensuse.org/package/binaries/home:fl0id/qesteidutil?repository=Fedora_26

**Comment by owocki** [comment]
@abitrolly would you mind rejecting the current claimee's claim at https://gitcoin.co/funding/details?url=https://github.com/open-eid/linux-installer/issues/13 ? since they no-showed, id love to let @step21 claim it

**Comment by abitrolly** [comment]
@owocki does rejecting cancel funding? The interface is not clear about it.

**Comment by abitrolly** [comment]
The issue is free to claim, @step21. )

I looked if I can package it myself, but I couldn't find sufficiently short tutorial or at least a picture with process overview get started. All I understood is that I need a .spec file. Then there are lengthy guidelines, which carry too much bloody details.

If you already experienced with packaging, maybe https://github.com/jordansissel/fpm would be a better way.

**Comment by step21** [comment]
Thanks! I saw, but yesterday I had some problems to claim it as described here: https://github.com/MetaMask/metamask-extension/issues/1081#issuecomment-355372620
Did you see the rpm link I posted, did that work? You can download it from there directly, but you can also add it as a repo with this url https://download.opensuse.org/repositories/home:/fl0id/Fedora_26/home:fl0id.repo (as stated above, others are also working independently to bring it back to the official repos, but no idea how long that will take)

**Comment by abitrolly** [comment]
@step21 I would be interested to see it officially resurrected in Fedora repo, where it could get more security reviews.

**Comment by step21** [comment]
This is being worked on independently, it doesn't really make sense or make it faster for me to duplicate that effort and this was not mentioned in the bounty/thread before. You can track the status of this here https://bugzilla.redhat.com/show_bug.cgi?id=1519323 (no idea how long it will take)
However please be advised, that this does not include (as for many packages in most distributions) any kind of security review. As long as it confirms to packaging guidelines etc it is as reasonably secure as any other program, but mostly upstream (the original authors) is responsible for writing secure code, as everywhere else.
Anyway, did you test the packages I built so far? I could run them, but I cannot test them properly as I do not have an effected ID card and even if I had I am pretty sure you can just update once.

**Comment by abitrolly** [comment]
@step21 I can only test binaries coming from official Fedora channels, sorry for not making it clear in the first place. This ticket and bounty is not about "compiling package for me". It is about supporting Fedora in proper way provided by this operating system, through its official channels.

There is also a security concern about running packages with critical software from other people servers, even if those people are with good intentions. Fedora builds are controlled, so that they are build from verified sources and no modification are made except those that are written in package spec. I can't be sure that your packages were not modified by you or any 3rd party, and don't want to risk my keys. Sorry about that.

**Comment by abitrolly** [comment]
I didn't notice https://bugzilla.redhat.com/show_bug.cgi?id=1519323 in the first place. There is no sense to duplicate efforts, and I wish we had noticed @Germano0 work faster, so he could claim the bounty.

**Comment by abitrolly** [comment]
@step21 I don't mind splitting the bounty if you can help resolve remaining issues in https://bugzilla.redhat.com/show_bug.cgi?id=1519323.

@owocki it is possible to split the bounty, right?

**Comment by Germano0** [comment]
If you are a Fedora user you can temporarily use my Copr repository https://copr.fedorainfracloud.org/coprs/germano/este-id-card-tools/
Why este id packages are not yet available in official Fedora repositories? Because Fedora packages guidelines are very strict and packages review take some amount of time.
During past weeks I have been very busy, but now I am resuming the package reviews process, for example I have just updated https://bugzilla.redhat.com/show_bug.cgi?id=1519747

**Comment by step21** [comment]
Yeah, that was not really clear. You still might consider my package, as on closer inspection the spec file in the bugzilla thread is only fedora 27, unless I am missing the part where there is one for 26.

I know that it is not only about 'compiling packages for you' but that is why I also asked multiple times what you would prefer (before and after I had seen the bugzilla report). As I have also said, I did not build the packages on my own server, but they are built by the opensuse build service, which is very established in the community and can be used to built software for all kinds of distributions, and is in no way specific to suse. (in fact I enabled only fedora packages) It creates a container with a fedora system, just for building, and then this container is discarded. As stated, I could also provide instructions to build it yourself, from some points of view that would be even more secure than using a fedora package, unless you do not trust the original authors but then well... better not use it :)
( @Germano0 I also indirectly mentioned your copr repository when linking to your comment in the other repository, but I guess they did not see it or like it )
We could also try to have it included with the original repository, so that maybe in the future open-eid could provide it themselves - as really, having it as a fedora package does not make it more secure. Whether they would be open for that I do not know.

That said, if claiming works again for me and I try to push this to get included with/for fedora 26, I would like the full bounty, as I only saw the bugzilla thread when my package was already building on the build service, overall I spent quite some time on this already and I assumed your priority would be updating your card as fast as possible, instead of waiting for a package included with fedora, which by most accounts, does not make it more secure in itself. Lastly, with transaction fees as they are, it makes even less sense to split I think as I might pay up to 50 % as transactions fees and then I would just rather leave it I think.

**Comment by Germano0** [comment]
I really have no time to follow the whole discussion, so I can shortly summarise:
- at the moment, due lack of free time to accomplish particular requests (or even discuss them) I state that I do not have any will of getting any part of the bounty;
- All Estonia ID software packages will return back in Fedora official repositories for versions >= 27 because on older Fedora versions there are various problems with OpenSSL library that has been deprecated on recent Estonia ID software releases. For the same reason there is no plan for EPEL;
- anybody wanting to help, please [join the Fedora package collection maintainers](https://fedoraproject.org/wiki/Join_the_package_collection_maintainers)

Best regards

**Comment by abitrolly** [comment]
Thank you @Germano0 for settling things up. We are still experimenting with this bounty system, and it will take some time to bring up mechanics that makes problem solving fun for all participants.

@step21 I apologize again that I missed links to bugzilla that you've sent in previous posts. With rolling update model of Fedora, I don't think it is valuable to port to 26. But making .spec files native for this repository with instruction how to update/maintain this package in Fedora will make it fair to claim for full bounty. I think neither me nor @Germano0 mind if Fedora will get more collection maintainers thanks to your contribution. :)

**Comment by step21** [comment]
Ok, cool. By native you mean on the fedora bugzilla or in the github source?  (like for example debian spec like files are included there)

**Comment by abitrolly** [comment]
By native I mean that this Github repo will contain all necessary files in `fedora/` subdir along with instructions in README.md to build and install Fedora package from scratch. Like Debian, right.

To avoid bitrot this repo should also contain instructions how to compare packaging files with files in Fedora repo and explanation of how to sync those files. This requires description of what steps are required to push changes on Fedora side. The doc may go into separate file.

**Comment by gitcoinbot** [comment]
__The funding of 0.045 ETH (45.85 USD) attached has been claimed by @step21.__ 

 @step21, please leave a comment to let the funder (@abitrolly) and the other parties involved your implementation plan.  If you don't leave a comment, the funder may expire your claim at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/funding/details?url=https://github.com/open-eid/linux-installer/issues/13)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $14514.91 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by owocki** [comment]
> @owocki does rejecting cancel funding? The interface is not clear about it.

sorry for the radio silence @abitrolly .  was away skiing this weekend.  

not it doesnt.  it just rejects the current claim.  i will update the interface.. thanks!

**Comment by owocki** [comment]
@step21 👋  good to have you here :)

**Comment by step21** [comment]
Hi @abitrolly I submitted a PR now to the qesteidutil. https://github.com/open-eid/qesteidutil/pull/89
Please let me know what you think. I tried to make the instructions clear and not assume too much, but of course not everything can be covered.

**Comment by step21** [comment]
Everything is reviewed now and will hopefully be merged soon.

**Comment by vs77bb** [comment]
@step21 Fantastic! cc @owocki

**Comment by step21** [comment]
Merged now

**Comment by owocki** [comment]
@abitrolly mind releasing the funds to @step21  at https://gitcoin.co/funding/details?url=https://github.com/open-eid/linux-installer/issues/13 ?

**Comment by abitrolly** [comment]
There is still no package built for F28 at https://apps.fedoraproject.org/packages/esteidcerts/ and all tickets on Bugzilla are still open, so @step21 if you can help @Germano0 with upstream review by notifying https://bugzilla.redhat.com/show_bug.cgi?id=1518957 that pagure issue is resolved and find out how to schedule F28 built - that will help to move it forward.

**Comment by gitcoinbot** [comment]
⚡️ A tip worth 0.045  ETH ($52.0) has been granted to @germano0 for this issue. ⚡️ 

Nice work @germano0, check your email for further instructions. 

 * $45877.14 in Funded OSS Work Available at: https://gitcoin.co/explorer
 * Incentivize contributions to your repo: <a href='https://gitcoin.co/tip'>Send a Tip</a> or <a href='https://gitcoin.co/funding/new'>Fund a PR</a>
 * No Email? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>

**Comment by gitcoinbot** [comment]
__The funding of 0.045 ETH (53.03 USD) attached to this issue has been approved & issued to @step21.__  

 * Learn more at [on the gitcoin issue page](https://gitcoin.co/funding/details?url=https://github.com/open-eid/linux-installer/issues/13)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $46728.28 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by abitrolly** [comment]
Sorry for F28 reference. I meant F27 of course. Now that F27 packages are officially being built and delivered, I'd sent tips to @Germano0 and approved claim by @step21 for working on the issue.

@step21 it would be nice if I you could make a PR that will cover how to install this soft on Fedora 27 from official repos. Maybe merge wiki into README/docs, update install script to be compatible with Fedora, or merge RPM files into this dir so that this software could be built independently.

I still haven't figured out how to install the software. Will start trying with `esteidcerts` later.

**Comment by step21** [comment]
Thank you for this. After the last post, I was indeed a bit confused, and not sure what to do. Because of the version but also because it seemed to become a never ending issue (new version, new package (new bugzilla thread), new task after having asked before if PR with instructions would cover the issue) and because I didn't really want to get involved in any disagreement between @Germano0 and Fedora.
I can add a note to the qesteidutils package where I added the other things. I do not think this repo (linux-installer) is appropriate, as it explicitly states that it is for Ubuntu/Debian. For other distributions (gentoo etc) there is a seperate repo, but I am not sure if there should be one for Fedora/rpm, in my opinion as long as rpms are now built and spec files are part of source repos, this should not be necessary.
You would install it from official repos exactly as any other software. I assume you are used to Fedora and how it installs software. How did you install the previous versions?

**Comment by metsma** [comment]
> I still haven't figured out how to install the software. Will start trying with esteidcerts later.
Forget esteidcerts, its was used by libdigidoc and witch is not ported to openssl 1.1 (upstream has not confirmed that they will)

**Comment by Germano0** [comment]
In two weeks we should be able to have QDigiDoc fully working on Fedora >= 27.
There are mandatory timings for testing packages before getting them into stable repository.
qesteidutil will come soon too. I don't have an estimated time for it

**Comment by Germano0** [comment]
Just sent created qdigidoc update https://bodhi.fedoraproject.org/updates/qdigidoc-3.13.4-1.fc27
and now I am finishing the review of qesteidutil

**Comment by Germano0** [comment]
Released qdigidoc in Fedora testing repositories.
Could you please test it https://bodhi.fedoraproject.org/updates/FEDORA-2018-569f4b05f7
then login with you Fedora Account System account and leave a karma feedback?
Thank you

**Comment by abitrolly** [comment]
Installed.

    sudo dnf install qdigidoc --enablerepo=updates-testing

Could not find executable `qdigidoc`.

    ➜  ~ qdigidoc
    zsh: qdigidoc: command not found...

After reading internets, found `qdigidocclient`. Maybe rename it to just `qdigidoc`?

Launching client shows console error and warning window.
```
➜  ~ qdigidocclient
QObject: Cannot create children for a parent that is in a different thread.
(Parent is QSigner(0x5563958160a0), parent's thread is QThread(0x5563954459c0), current thread is QSigner(0x5563958160a0)
Loading: "opensc-pkcs11.so"
"OPENSC PROJECT                   (2.20)" 
"OPENSC SMARTCARD FRAMEWORK       (0.17)" 
Flags: 0
ERROR [TSL.cpp:248] - TSL tl-mp.xml status: TSL is expired
DEBUG [Connect.cpp:53] - Connecting to URL: https://ec.europa.eu/information_society/policy/esignature/trusted-list/tl-mp.xml
DEBUG [Connect.cpp:75] - Connecting to Host: ec.europa.eu:443
DEBUG [TSL.cpp:503] - TSL Signature is signed with untrusted certificate
ERROR [TSL.cpp:283] - TSL tl-mp.xml signature is invalid
```



![image](https://user-images.githubusercontent.com/8781107/35908095-b9771bfe-0c00-11e8-8edb-72cf21ac724f.png)

Link leads to https://www.id.ee/?id=37012

And as you see on the screenshot it doesn't recognize my reader or card. Running under `sudo` doesn't help.
```
➜  ~ sudo qdigidocclient
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
No protocol specified
qt.qpa.screen: QXcbConnection: Could not connect to display :0
Could not connect to any X display.
```
dmesg
```
[21445.212565] usb 2-1: new full-speed USB device number 4 using xhci_hcd
[21445.345001] usb 2-1: New USB device found, idVendor=076b, idProduct=a021
[21445.345006] usb 2-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[21445.345009] usb 2-1: Product: Smart Card Reader
[21445.345012] usb 2-1: Manufacturer: USB
```

**Comment by Germano0** [comment]
@abitrolly I am going to update the qdigido cclient to lastest release
Meanwhile,

1. are you using Wayland or Xorg?
2. Please show us the output of `# systemctl status pcscd`

**Comment by abitrolly** [comment]
@Germano0 Wayland
```
➜  ~ sudo systemctl status pcscd
● pcscd.service - PC/SC Smart Card Daemon
   Loaded: loaded (/usr/lib/systemd/system/pcscd.service; indirect; vendor preset: di
   Active: inactive (dead)
```

**Comment by Germano0** [comment]
@abitrolly 
- concerning Wayland errors, @metsma can tell us more
- card reader: running once `systemctl enable --now pcscd` should enable your system to correctly detect card readers

**Comment by abitrolly** [comment]
@Germano0 got it running. :) Is it possible to add this `pcscd` service startup as a requirement for `qdigidoc`?

The only missing part right now is `qesteidutil` required to make checks for updating ID card.

![screenshot from 2018-02-07 14-34-31](https://user-images.githubusercontent.com/8781107/35914425-4ae7a316-0c14-11e8-9533-74b129528456.png)

**Comment by Germano0** [comment]
> Is it possible to add this pcscd service startup as a requirement for qdigidoc?
No [it is forbidden by Fedora policy](https://fedoraproject.org/wiki/Packaging:DefaultServices?rd=DefaultServices#Restrictions). Rather  psdcd Fedora package maintainer could enable such service when user plugs a card reader or when the system has already a card reader https://fedoraproject.org/wiki/Packaging:Systemd#Hardware_activation

**Comment by Germano0** [comment]
New qdigidoc update https://bodhi.fedoraproject.org/updates/qdigidoc-3.13.5-1.fc27

**Comment by metsma** [comment]
Did you patch the build
https://github.com/open-eid/qdigidoc/wiki/DeveloperTips#building-in-sandboxed-environment

**Comment by Germano0** [comment]
@metsma 
> Did you patch the build
> https://github.com/open-eid/qdigidoc/wiki/DeveloperTips#building-in-sandboxed-environment

The Fedora patch was obsolete. I will update it as soon as possible. Thank you

**Comment by Germano0** [comment]
This time qdigidoc Fedora build should be perfect
https://bodhi.fedoraproject.org/updates/qdigidoc-3.13.5-3.fc27

**Comment by Germano0** [comment]
[qdigidoc update](https://bodhi.fedoraproject.org/updates/FEDORA-2018-8d148ceb02)
[qesteidutil update](https://bodhi.fedoraproject.org/updates/FEDORA-2018-54c7a5b01b)
If you login with FAS account (Fedora Account System) and leave karma feedback you will help in pushing the updates into stable repository as soon as possible

**Comment by abitrolly** [comment]
`qdigidoc` starts ok now. It still doesn't see my card without explicitly running `pcscd` service. Is it possible to use https://fedoraproject.org/wiki/Packaging:Systemd#Socket_activation for starting it if hardware activation is not implemented?

**Comment by Germano0** [comment]
I don't know, it is a problem that pscd maintainer should care of

**Comment by abitrolly** [comment]
The `qesteidutil` starts ok and I can see my data. But.. I don't see Update button on the interface like it is shown at https://youtu.be/BRGFSSoW-Xc?t=9 and @eesti.ee e-mail is grayed out and hitting Certificates key does nothing. Is it expected behaviour? There is a suspicious `Failed to parse public key` message in console.
```
➜  ~ qesteidutil 
libpng warning: iCCP: known incorrect sRGB profile
Failed to parse public key
```

https://www.politsei.ee/id-kaardi-kontroll/en says that I need to update my document.

![image](https://user-images.githubusercontent.com/8781107/36242347-75c78650-122c-11e8-9994-5be1b1fddeae.png)

**Comment by metsma** [comment]
Can you send diagnostics output

**Comment by abitrolly** [comment]
@metsma `qesteidutil` doesn't respond to command line options, so I copy/pasted this from GUI window.
```
Locale: English / en_US.UTF-8

Application version: 3.12.10.0 (64 bit)
OS: 
CPU: Intel(R) Core(TM) i5-4300U CPU @ 1.90GHz
Kernel: Linux 4.14.18-300.fc27.x86_64 #1 SMP Thu Feb 8 01:35:31 UTC 2018 x86_64

Libraries:
qdigidoc (3.13.5)
qesteidutil (3.12.10)
openssl (1.1.0g)
pcsc-lite (1.8.22)
opensc (0.17.0)
QT (5.9.4)

Arguments: qesteidutil
Library paths: /usr/lib64/qt5/plugins;/usr/bin
URLs:
CONFIG_URL: https://id.eesti.ee/config.json

Central Configuration:

Smart Card service status: Running
Smart Card readers:
OMNIKEY AG Smart Card Reader 00 00 max APDU size 0
Reader state: PRESENT, INUSE
ATR cold - 3BFA1800008031FE45FE654944202F20504B4903
ATR warm - 3BFE1800008031FE45803180664090A4162A00830F9000EF
ID - 38206070147


Browsers:
firefox (58.0.1)

USB info:
Bus 001 Device 002: ID 8087:8000 Intel Corp. 
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 003 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 002 Device 003: ID 04ca:703c Lite-On Technology Corp. 
Bus 002 Device 007: ID 076b:a021 OmniKey AG CCID Smart Card Reader
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

```

**Comment by Germano0** [comment]
It looks like not all config.* files have been updated on Fedora repository, despite I did that. Probably I missed some errors during the procedure. I am releasing a new build right now

**Comment by abitrolly** [comment]
Yep. I was just looking why OS field is not filled there. Looks like Fedora detection should occur at https://github.com/open-eid/qt-common

**Comment by Germano0** [comment]
> It looks like not all config.* files have been updated on Fedora repository, despite I did that. Probably I missed some errors during the procedure. I am releasing a new build right now

I was wrong, config.json config.pub config.rsa sha256 checksum is identical to the ones from the wiki. I do not know what could be the problem
https://src.fedoraproject.org/rpms/qesteidutil/tree/master

> Yep. I was just looking why OS field is not filled there. Looks like Fedora detection should occur at https://github.com/open-eid/qt-common

???

**Comment by metsma** [comment]
Look here https://github.com/open-eid/qesteidutil/tree/master/fedora
According this patch is not needed

**Comment by abitrolly** [comment]
> ???

Code for Diagnostic is maintained in that repo - https://github.com/open-eid/qt-common

**Comment by Germano0** [comment]
@metsma 
> Look here https://github.com/open-eid/qesteidutil/tree/master/fedora

It contains incomplete and unreliable instructions

> According this patch is not needed

Which patch file??

**Comment by metsma** [comment]
Please fix the instructions then

Patch0:         sandbox-compilation.patch

**Comment by Germano0** [comment]
Problem solved,
[%patch0 -p1](https://src.fedoraproject.org/rpms/qesteidutil/blob/master/f/qesteidutil.spec#_39)
was missing.
To test the new version
$ bodhi updates download --updateid FEDORA-2018-54c7a5b01b
# dnf update downloaded_file.rpm

If you login with FAS account (Fedora Account System) and leave karma feedback you will help in pushing the updates into stable repository as soon as possible and we can close this ticket.
https://koji.fedoraproject.org/koji/taskinfo?taskID=25068324


@metsma 
> Please fix the instructions then
> 
> Patch0: sandbox-compilation.patch
Ask the author of that guide @step21

**Comment by abitrolly** [comment]
![image](https://user-images.githubusercontent.com/8781107/36272541-e04ba4f0-1292-11e8-9706-b08c1797c3bc.png)

👍 :)

**Comment by Germano0** [comment]
Now I am starting to taking care of browser side of Estonia ID card software stack. I created 3 personal repositories to work on RPM spec files. Don't mind the content because it is in a very early  stage
https://pagure.io/estonianidcard_spec
https://pagure.io/firefox-pkcs11-loader_spec
https://pagure.io/webextension-token-signing_spec

**Comment by metsma** [comment]
can you rename estonianidcard to open-eid

**Comment by Germano0** [comment]
> can you rename estonianidcard to open-eid

Sure, I updated previous comment

**Comment by abitrolly** [comment]
I successfully updated my card certificates. Thanks everyone. =)

**Comment by Germano0** [comment]
> Now I am going to start to taking care of browser side of Estonia ID card software stack. I created 3 personal repositories to work on RPM spec files. Don't mind the content because it is in a very early stage
> https://pagure.io/open-eid_spec (meta package to install the whole software stack)
> https://pagure.io/firefox-pkcs11-loader_spec
> https://pagure.io/webextension-token-signing_spec

The browser side is almost finished.

I am very glad to announce that open-eid stack will be released into [EPEL7](https://fedoraproject.org/wiki/EPEL/FAQ) the official additional community repository for Red Hat Enterprise Linux / CentOS / Scientific Linux !

**Comment by metsma** [comment]
Requires:       esteidcerts  << not needed

**Comment by Germano0** [comment]
> Requires: esteidcerts << not needed

Could you please explain? Thank you

**Comment by metsma** [comment]
Browser packages do not need it and all Estonian CA-s are already trusted by Browsers

**Comment by Germano0** [comment]
> Browser packages do not need it and all Estonian CA-s are already trusted by Browsers

Changes applied, thank you!

**Comment by Germano0** [comment]
@metsma concerning chrome-token-signing, do you confirm that binary /usr/bin/chrome-token-signing is the component that handles PIN prompt and it is needed by both Firefox and Chrome browsers?

**Comment by metsma** [comment]
Pin prompt, certification choice and communication with smartcard

**Comment by Germano0** [comment]
[Review request: open-eid - Meta-package for Estonian Electronic Identity Software](https://bugzilla.redhat.com/show_bug.cgi?id=1550743)

[Review request: webextension-token-signing - Firefox extension for signing with your eID on the web](https://bugzilla.redhat.com/show_bug.cgi?id=1550740)

**Comment by Germano0** [comment]
**TESTERS WANTED**
I am currently working on investigating what is not working in Fedora package [chrome-token-signing](https://github.com/open-eid/chrome-token-signing) which on Fedora is called [webextension-token-signing](https://src.fedoraproject.org/rpms/webextension-token-signing)
I am experiencing failures during documents signing on https://www.digidoc.ee/ and during tests on https://hwcrypto.github.io/hwcrypto.js/sign.html and https://hwcrypto.github.io/hwcrypto.js/sign.html
If you want to test it you can simply install the whole Estonia ID card software stack with command

`# dnf --enablerepo=updates-testing install open-eid`

Any suggestion and contributions are welcomed.
Thank you!

**Comment by Germano0** [comment]
> TESTERS WANTED (Firefox only)
> I am currently working on investigating what is not working in Fedora package chrome-token-signing which on Fedora is called webextension-token-signing
> I am experiencing failures during documents signing on https://www.digidoc.ee/ and during tests on https://hwcrypto.github.io/hwcrypto.js/sign.html and https://hwcrypto.github.io/hwcrypto.js/sign.html
> If you want to test it you can simply install the whole Estonia ID card software stack with command
> 
> # dnf --enablerepo=updates-testing install open-eid
> 
> Any suggestion and contributions are welcomed.
> Thank you!

Fix coming soon
https://github.com/open-eid/chrome-token-signing/issues/80#issuecomment-372034703

**Comment by Germano0** [comment]
The work is almost finished on both Fedora and EPEL (RHEL / CentOS / Scientific Linux) repositories.
I just submitted the packages to updates testing repositories, and in next days I will send a new message with instructions about how to test them and provide feedbacks.

Best regards

**Comment by Germano0** [comment]
A quick resume about the testing process


**Fedora situation**
Certain packages are still in updates-testing repository. In order to help pushing the update to stable repository, open hereunder Fedora 27: package release process page, register and provide karma feedback
Everything works fine, you may need to manually start pcscd service while using open-eid on Firefox. [I am working on fixing this too](https://bugzilla.redhat.com/show_bug.cgi?id=1545027)

To install:
 dnf install --enablerepo=updates-testing open-eid

To start pcscd service:
 systemctl start pcscd

To permanently enable pcscd service:
 systemctl enable pcscd


***CentOS / Red Hat Enterprise Linux / Scientific Linux situation***
open-eid provided by official repository Fedora EPEL (Extra Packages for Enterprise Linux)
Certain packages are still in updates-testing repository. In order to help pushing the update to stable repository, open hereunder Fedora EPEL 7: package release process page, register and provide karma feedback.
qdigidoc and qesteidutil work fine, you may need to manually start pcscd service.
Firefox is not yet completely supported because firefox-pkcs11-loader is not compatible with Firefox ESR 52. [You may need to wait at least for May month](https://bugzilla.redhat.com/show_bug.cgi?id=1561933)

To install:
 yum install epel-release
 yum install --enablerepo=epel-testing open-eid

To start pcscd service:
 systemctl start pcscd

To permanently enable pcscd service:
 systemctl enable pcscd


**List of open-eid packages**

**open-eid**
Meta package that installs all following packages
[Fedora 27: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-2018-e84d057947)
[Fedora EPEL 7: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2018-e7b4a7b2ad)


**esteidcerts**
[Github](https://github.com/open-eid/libdigidocpp)
[Fedora: package page](https://src.fedoraproject.org/rpms/esteidcerts)
[Fedora 27: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-2018-913daabed0)
[Fedora EPEL 7: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2018-7d396fb6e6)


**libdigidocpp**
La libreria libdigidocpp fornisce funzioni di creazione, firma e verifica di documenti firmati digitalmente. La libreria opera in base agli standard XAdES e XML-DSIG 
[Documentazione](http://open-eid.github.io/libdigidocpp/)
[Github](https://github.com/open-eid/libdigidocpp)
[Fedora: package page](https://src.fedoraproject.org/rpms/libdigidocpp)
[Fedora 27: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-2018-77221a1e52)
[Fedora EPEL 7: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2018-18e8bc4375)


**firefox-pkcs11-loader**
Fornisce uno script + una webextension che configurano Firefox per l'utilizzo dei protocolli pkcs11, evitando all'utente di dover configurare a mano il browser. Una volta installato abilita il browser all'utilizzo anche di dispositivi di autenticazione di terze parti
[Github](https://github.com/open-eid/firefox-pkcs11-loader)
[Fedora: package page](https://src.fedoraproject.org/rpms/firefox-pkcs11-loader)
[Fedora 27: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-2018-f77abb067c)
Fedora EPEL 7: still not compatible

**chrome-token-signing**
Pacchetto Chrome e Firefox contenente una webextension per permettere di apporre firme digitali sul web utilizzando le carte estoni. La parte in qt lib serve per la richiesta del PIN ed altre funzioni basilari
[Documentazione](https://github.com/open-eid/chrome-token-signing/wiki)
[Github](https://github.com/open-eid/chrome-token-signing)
[Fedora: package page](https://src.fedoraproject.org/rpms/webextension-token-signing)
[Fedora 27: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-2018-6dcee533d9)
[Fedora EPEL 7: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2018-18fc85ed97)

**qesteidutil**
Software che permette il controllo delle carte elettroniche estoni, la validità dei certificati, la manutenzione dei codici PIN e PUK.
[Github](https://github.com/open-eid/qesteidutil)
[Fedora: package page](https://src.fedoraproject.org/rpms/qesteidutil)
[Fedora 27: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-2018-54c7a5b01b)
[Fedora EPEL 7: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2018-15ee907aef)

**qdigidoc**
qdigidoc è la versione 3 di DigiDoc Client, un software che permette di apporre o verificare firme digitali di documenti, aprire e salvare documenti che sono dentro un file contenitore firmato digitalmente
[Github](https://github.com/open-eid/qdigidoc)
[Fedora: package page](https://src.fedoraproject.org/rpms/qdigidoc)
[Fedora 27: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-2018-8d148ceb02)
[Fedora EPEL 7: package release process page](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2018-f5d2575e9f)

**Comment by metsma** [comment]
please remove esteidcerts, it is unused, obsolete, deprecated and with outdated certs.

**Comment by Germano0** [comment]
> please remove esteidcerts, it is unused, obsolete, deprecated and with outdated certs.

Done. Removed esteidcerts package as Requires: from [firefox-pkcs11-loader](https://src.fedoraproject.org/rpms/firefox-pkcs11-loader/c/cc716af5fc8f4f209cea751e332bb545374cdc37?branch=master) and [chrome-token-signing](https://src.fedoraproject.org/rpms/webextension-token-signing/c/edd3272b813993eafee80ee2b722f7b61242de70?branch=master)

**Comment by Germano0** [comment]
Work complete.
Concerning web support on Enterprise Linux 7, you have to wait for a newer Firefox ESR version that will support web extensions. This should happen in ~May / June.
Best regards

**Comment by smirnov** [comment]
Hmmm, I am unable to run the qdigidocclient as it still throws error: qdigidocclient: symbol lookup error: /lib64/libdigidocpp.so.1: undefined symbol: _ZN11xalanc_1_1116XalanTransformer10initializeERN11xercesc_3_113MemoryManagerE

Is this expected behaviour?

**Comment by Germano0** [comment]
> Hmmm, I am unable to run the qdigidocclient on Fedora 28, as it still throws error:
> 
> qdigidocclient: symbol lookup error: /lib64/libdigidocpp.so.1: undefined symbol: _ZN11xalanc_1_1116XalanTransformer10initializeERN11xercesc_3_113MemoryManagerE
> 
> Is this expected behaviour?

I will post update messages on fixing F28 beta qdigidoc, on the proper Fedora bugreport
https://bugzilla.redhat.com/show_bug.cgi?id=1573162

**Comment by metsma** [comment]
libdigidocpp is compiled against xalan-c package and maybe xml-security-c removed xalan-c dependency.
Remove xalan-c-devel package when compiling libdigidocpp

**Comment by abitrolly** [comment]
To summarize. On Fedora 30+ the EID software is called **qdigidoc4** and is available from main official Fedora repository from **`qdigidoc`** package

```
$ sudo dnf install qdigidoc
$ qdigidoc4
```

https://apps.fedoraproject.org/packages/qdigidoc/
